src/Voron/Data/CompactTrees/CompactKey.cs

```diff
@@ -17,12 +18,15 @@ namespace Voron.Data.CompactTrees;
 // when the same key is reused on multiple operations in the same transaction.
 public unsafe class CompactKey : IDisposable
 {
-    public readonly LowLevelTransaction Owner;
+    private static readonly ArrayPool<byte> StoragePool = ArrayPool<byte>.Shared;
+    private static readonly ArrayPool<long> KeyMappingPool = ArrayPool<long>.Shared;
```

Comment:
why do this? They are the same instances, after all

Reply:
One is long and the other one is bytes... I want to avoid having to Unsafe access the long array as size of byte.

Reply:
I was actually talking about why you are making a `static field` for each of those, where the _is_ ` static field` for them already in the `Shared`